.dockerignore

```diff
@@ -11,5 +11,6 @@
 !build_rust.py
 
 rust/target
+synapse/*.so
```

Comment:
This is a very sensible idea, but I don't understand why things break without it. Shouldn't the `pip install` steps on lines 
131 and 133 of the dockerfile have rebuilt a correct .so for that platform?

Reply:
If you are doing dev and do a `poetry install` at some point, the .so is generated but on the host, meaning it's bind to the glibc version of the host.
Then the build system is configured to skip the Rust build if the code hasn't change => the .so is not generated again from inside the container since it is already around and the code hasn't changed.